app/controllers/chat_controller.rb

```diff
@@ -429,13 +429,17 @@ def move_messages_to_channel
   end
 
   def flag
+    RateLimiter.new(current_user, "flag_chat_message", 4, 1.minutes).performed!
+
     permitted_params =
       params.permit(
         %i[chat_message_id flag_type_id message is_warning take_action queue_for_review],
       )
 
     chat_message =
-      ChatMessage.includes(:chat_channel).find_by(id: permitted_params[:chat_message_id])
+      ChatMessage.includes(:chat_channel, :revisions).find_by(
+        id: permitted_params[:chat_message_id],
+      )
     raise Discourse::InvalidParameters.new(:chat_message_id) if !chat_message
```

Comment:
How about we replace all of this by:

```
chat_message = ChatMessage.includes(...).find(id)
```

Reply:
That would return a different exception if not found. Let me check if it's handled the same way in the `ApplicationController`.

Reply:
I guess we could use `rescue_from` in this controller for example. Not sure it would be better though?